cloud/amazon/ec2_vpc_nat_gateway.py

```diff
@@ -0,0 +1,298 @@
+                    for nat_gateway_address in nat_gateway.get('NatGatewayAddresses'):
+                        if nat_gateway_address.get('AllocationId') == allocation_id:
+                            gateway_list.append({'state': nat_gateway.get('State'),
+                                                 'created': nat_gateway.get('CreateTime'),
+                                                 'nat_gateway_id': nat_gateway.get('NatGatewayId')})
+                return gateway_list
+        return [{'state': 'absent'}]
+    except botocore.exceptions.ClientError as ce:
+        if "NatGatewayNotFound" in ce.message:
+            return [{'state': 'absent'}]
+        else:
+            module.fail_json(msg=ce.message)
+
+
+def delete_nat_gateway(ec2_client=None, module=None, nat_gateway=None):
+    """ Delete an existing NAT gateway """
+    nat_gateway_address = nat_gateway.get('NatGatewayAddresses')[0]
+    nat_gateway_id = nat_gateway['NatGatewayId']
+    results = dict(changed=True, nat_gateway_id=nat_gateway_id,
+                   public_ip=nat_gateway_address.get('PublicIp'),
+                   private_ip=nat_gateway_address.get('PrivateIp'),
+                   allocation_id=nat_gateway_address.get('AllocationId'))
+    ec2_client.delete_nat_gateway(NatGatewayId=nat_gateway_id)
+    wait = module.params.get('wait')
+    if wait:
+        wait_timeout = time.time() + module.params.get('wait_timeout')
+        while wait_timeout > time.time():
+            nat_gateway_status_list = get_nat_gateway_status_list(ec2_client=ec2_client, module=module)
+            if nat_gateway_status_list[0].get('state') in ('deleted', 'absent'):
+                module.exit_json(**results)
+            else:
+                time.sleep(5)
+        module.fail_json(msg="Waited too long for VPC NAT Gateway to be deleted.")
+    else:
+        module.exit_json(**results)
+
+
+def create_nat_gateway(ec2_client=None, module=None):
+    """ Create the NAT gateway """
+    try:
+        result = ec2_client.create_nat_gateway(SubnetId=module.params.get('subnet_id'),
+                                               AllocationId=module.params.get('allocation_id'))
+        wait = module.params.get('wait')
+        start_time = datetime.datetime.utcnow()
+        if wait:
+            wait_timeout = time.time() + module.params.get('wait_timeout')
+            while wait_timeout > time.time():
+                present_status_list = get_nat_gateway_status_list(ec2_client=ec2_client, module=module)
+                for present_status in present_status_list:
+                    create_time = present_status.get('created')
+                    if present_status.get('state') == 'failed' and create_time.replace(tzinfo=None) >= start_time:
+                        module.fail_json(msg="Failed to create VPC NAT Gateway")
+                    elif present_status.get('state') == 'available':
+                        nat_gateway_id = result['NatGateway']['NatGatewayId']
```

Comment:
Have a look at using the helper function from module utils "camel_dict_to_snake_dict" and return all the parameters that boto3 returns not just gateway id - i guarantee someone will want one of them later!